Neos.ContentRepository/Classes/Migration/Service/GeneratorService.php

```diff
@@ -0,0 +1,60 @@
+<?php
+namespace Neos\ContentRepository\Migration\Service;
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Package\Exception\UnknownPackageException;
+use Neos\Flow\Package\PackageManager;
+use Neos\Utility\Exception\FilesException;
+use Neos\Utility\Files;
+
+/**
+ * Service for the Migration generator
+ *
+ */
+class GeneratorService
+{
+
+    /**
+     * @Flow\Inject
+     * @var PackageManager
+     */
+    protected PackageManager $packageManager;
+
+    /**
+     * Creates a node migration for the given $packageKey
+     *
+     * @param string $packageKey the package key
+     * @return string
+     * @throws UnknownPackageException
+     * @throws FilesException
+     */
+    public function createNodeMigration(string $packageKey): string
```

Comment:
Hi ;)

i think the method has too many resposibilities (imagine you would try to test this service (currently impossible ^^))
we should find a more fitting name or better: handle the most "scary" sideeffect `file_put_contents` externally.

Btw it will be easier to test the  $timeStamp by injecting flows `Now` (instead of new \DateTimeImmutable()) which is a singleton and can be controlled in tests.

Maybe you can help us too ^^ @fcool

Reply:
I agree that the name of the method should be adjusted maybe `generateBoilerplateMigrationFileInPackage(string $packageKey)`.